pkg/csi/service/vanilla/controller.go

```diff
@@ -1825,14 +1829,24 @@ func (c *controller) processQueryResultsListVolumes(ctx context.Context, startin
 				}
 				// Populate List Volumes Entry Response
 				entries = append(entries, entry)
+				if volCounter == maxEntries {
+					nextTokenCounter = i + 1
+					break
+				}
 			}
 		}
 	}
 
-	// if length of queryAll entries > queryLimit, set nextToken to
-	// start_token + queryLimit
-	if len(cnsVolumes) > endIndex {
-		nextTokenInt := endIndex
+	// for cases where number of volumes are fewer than maxEntries.
+	if nextTokenCounter == 0 {
+		// After processing all cns volumes fewer than 'maxEntries' no. of volumes were found.
+		// and i would be len(cnsVolumes).
+		// So nextTokenCounter should be set to len(cnsVolumes)
+		nextTokenCounter = len(cnsVolumes)
+	}
+	// If there are more volumes to process, set the nextToken. Else, return empty nextToken
+	if len(cnsVolumes) > nextTokenCounter {
+		nextTokenInt := nextTokenCounter
```

Comment:
can we eliminate this step and directly evaluate as `nextToken = strconv.Itoa(nextTokenCounter)` ?

Reply:
Done